ember-cli/ember-fetch
# Convert to v2, addon. Just wrap `fetch`.

This PR is (what I think is) the goal. 
We can decide if we want to move towards this goal incrementally or deciding that this is what we want and skip all the in-between.


To move incrementally, we'd need to resolve the following PRs:
- https://github.com/ember-cli/ember-fetch/pull/753
- https://github.com/ember-cli/ember-fetch/pull/743
- https://github.com/ember-cli/ember-fetch/pull/742
- https://github.com/ember-cli/ember-fetch/pull/741
- https://github.com/ember-cli/ember-fetch/pull/732

Which... we can do -- I'm not sure it's worth keeping CI green through all that though. `ember-fetch` is in a bit of disrepair.

Support:
- 3.28+ (easily), maybe 3.4+
- ember-auto-import @ v2+ or embroider
- `@ember/test-waiters` v3+
- any node (node doesn't matter anymore)
- continue the renamed-modules support (import from fetch)

I'm not sure what the fastboot support story would be, but Node natively has `fetch`, `AbortController`, etc.

Dropped Support:
- Configuration: PREFER_NATIVE, NATIVE_PROMISE
- jQuery AJAX
- You no longer import AbortController. `AbortController` exists on `globalThis`
- You no longer import Headers. `Headers` exists on `globalThis`
- You no longer import Response. `Response` exists on `globalThis`



Since `ember-auto-import` tests against 3.4.0, and ember-auto-import is the only real requirement for this change, there is **no change** in supported ember-source.

If anything, the constraining function is `@ember/test-waiters`

Changes:
- [x] delete the old addon
- [x] create new fetch export
- [x] copy over old utils as separate exports
  - [x] remove './types', because these exist all natively
  - [x] remove './utils/determine-body-promise' because it used jQuery AJAX options
  - [x] remove './utils/mung-fetch-options' because it used jQuery AJAX options
- [x] **no build -- folks can just point at git to test this**
- [x] separate test app
- [x] yarn -> pnpm

After this PR:
- Add release plan
- remove release-it
